base/stream.jl

```diff
@@ -555,7 +555,9 @@ function uv_readcb(handle::Ptr{Cvoid}, nread::Cssize_t, buf::Ptr{Cvoid})
         if nread < 0
             if nread == UV_ENOBUFS && nrequested == 0
                 # remind the client that stream.buffer is full
-                notify(stream.readnotify)
+                if !isempty(stream.readnotify)
```

Comment:
this should already be the first line of `notify`, we shouldn't need to repeat this test twice in a row